rosenpass/src/protocol.rs

```diff
@@ -753,7 +856,122 @@ pub struct HandleMsgResult {
 }
 
 impl CryptoServer {
-    /// Respond to an incoming message
+    /// Process a message under load
+    /// This is one of the main entry point for the protocol.
+    /// Keeps track of messages processed, and qualifies messages using
+    /// cookie based DoS mitigation. Dispatches message for further processing
+    /// to `process_msg` handler if cookie is valid, otherwise sends a cookie reply
+    /// message for sender to process and verify for messages part of the handshake phase
+    /// (i.e. InitHello, InitConf messages only). Bails on messages sent by responder and
+    /// non-handshake messages.
+    pub fn handle_msg_under_load(
+        &mut self,
+        rx_buf: &[u8],
+        tx_buf: &mut [u8],
+        peer: PeerPtr,
+        socket_addr: SocketAddr,
+    ) -> Result<HandleMsgResult> {
+        //Check cookie value
+        let mut ip_addr_port = match socket_addr.ip() {
+            IpAddr::V4(ipv4) => ipv4.octets().to_vec(),
```

Comment:
I think it would be preferable to factor out IP-address specific functionality from protocol.rs; we want it to stay generic as opposed to relying on a specific network protocol…